Pass on the merging section of the manual (#1256)
tweag/nickel: doc/manual/merging.md

Merge is a **symmetric** operation (or, pedantically, commutative). In practice,
this means that order doesn't matter, and `left & right` is the same thing as
`right & left`. When the operands need to be distinguished, as we will see for

> default values for example, the idea is to use special annotations (metadata)
> instead of relying on the side of the operands.

Reviewer: ```suggestion
default values for example, the idea is to use special annotations (metadata)
instead of relying on the order of the operands.
```
I feel, "order" reads more naturally here.

When one or both of the common fields are not records, the merge will fail
unless one of the following condition hold:

- They are both of the same primitive data type (`Number`, `Bool`, `Enum`, or
  `String`) and they are equal

> - They are both arrays, and they are equal (checked by generating an application
>   of the lazy contract `contract.Equal`)

Reviewer: ```suggestion
- They are both arrays, and they are equal (checked by generating an application
  of the lazy contract `std.contract.Equal`)
```

Although the right-hand side `bar` doesn't have a definition, the resulting
`bar` field isn't optional anymore.

As long as an optional field doesn't have a value, it will be invisible to
record operations. Optional fields without a value don't show up in

> `std.record.fields`, it won't make `std.record.values` throw a missing field

Reviewer: ```suggestion
`std.record.fields`, they won't make `std.record.values` throw a missing field
```

definition error, etc.

```nickel
nickel> let Contract = {foo = 1, bar | optional}
nickel> std.record.values Contract
[ 1 ]

nickel> std.record.has_field "bar" Contract
false
```

> Optional field can still be discovered through metadata queries (run `nickel

Reviewer: ```suggestion
Optional fields can still be discovered through metadata queries (run `nickel
```

help query` or type `:help query` in the REPL for more information) or generated
documentation.

[^optional-with-value]: You can actually provide a value (default or not) for an optional
  field, but this nullifies the effect of `optional`: what you get is just a
  regular field

### Merge priorities

Priorities are specified using the `priority` annotation, followed by a number

> literal. There are also two other special priorities, the lowest priority
> `default`, and the higher priority `force` annotation.

Reviewer: ```suggestion
literal. There are also two other special priority annotations, the lowest priority
`default`, and the highest priority `force`. Both are written without the `priority` keyword.
```
